lib/superintendent/request/params_case_translator.rb

```diff
@@ -27,10 +27,31 @@ def underscored_keys(value)
       when Array
         value.map { |v| underscored_keys(v) }
       when Hash
+        if value['data'].present?
+          transform_type_to_snake_case(value['data'])
+        end
```

Comment:
`underscored_keys` is already recursive and handles arrays and hashes.
 You can just do this one line here instead of adding a function.
```
  value['type'] = underscored_key(value['type']) if value.has_key?('type')
```